packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -42,33 +41,39 @@ export class IpfsCredentialResolver implements CredentialResolver {
    *  provider,
    *  registrySettings,
    *  didStore );
-   * const credential = credentialResolver.getCredential('did:ethr:1234', 'sampleRole');
+   * const credential = credentialResolver.getCredential('did:ethr:1234', 'sampleRole', roleCredentialCache);
    * ```
    *
    * @param did subject DID for which the credential needs to be fetched
    * @param namespace role for which the credential needs to be fetched
+   * @param roleCredentialCache Cache to store role credentials
    * @returns
    */
   async getCredential(
     did: string,
-    namespace: string
+    namespace: string,
+    roleCredentialCache: IRoleCredentialCache
```

Comment:
I think it should be optional param. In this way, someone doesn't have to provide a cache if they don't want to use one. There is already the check for it in the method implementation
```suggestion
    roleCredentialCache?: IRoleCredentialCache | unknown
```

Reply:
I also had the same though but wasn't sure if it would be a good idea to make it optional. On the other hand I think it should be made mandatory since all the exported methods do perform `issuerVerification` some way or the other, so this would improve the efficiency for sure, but it could be optional and left it upto to the caller.

Reply:
Yes, on balance, I'm leaning towards it being optional